src/Plugin/Queue/Workers/Kafka/KafkaWorker.php

```
@@ -208,8 +215,13 @@ private function mapMessages(array $batch): array {
 	private function handleRow(array $message): array {
 		$row = [];
 		foreach ($this->fields as $fieldName => $fieldType) {
-			if (isset($message[$fieldName])) {
-				$row[$fieldName] = $this->morphValuesByFieldType($message[$fieldName], $fieldType);
+			$inputKeyName = $fieldName;
+			if ($this->customMapping !== []
+				&& isset($this->customMapping[$fieldName])) {
```

isset should be enough?

---

yes

---

In case isset is enough let's remove `$this->customMapping !== []` cuz it does not do anything, isset is enough

---

No, because during the construct, we use json_decode. If the JSON is corrupted, it can return false. That’s why we check if it’s an array.

---

You have declaration
```
private array $customMapping;
```

In case it returns false and we try to assign, we will fail already before this code, so I guess [] can be removed

```php
	$decodedMapping = json_decode($instance['custom_mapping'], true);
	$this->customMapping = $decodedMapping;
```
		
What if we will do this thing:

```php
	$decodedMapping = json_decode($instance['custom_mapping'], true) ?: []
```
	
So it will make sure we have array even if it returns false and avoid strict type fatal error

---

I’m not sure this is a good idea:
	1.	If we have a corrupted custom mapping, it’s more rational to throw an exception rather than proceed.
	2.	In most cases, we don’t have any mapping, so checking `$this->customMapping !== []` allows us to avoid consuming resources and performing unnecessary `isset` checks since we already know there’s no mapping.

---

1. If we want to throw an exception, it's better to do it explicitly. The main issue is about type annotation and that you cannot assign `false` to a property declared as `array`. This results in a fatal syntax error, which is easier to investigate than an exception
2. Regarding performance, checking '!== []' is more resource-intensive than using 'isset()'. The key point is that 'isset()' alone is sufficient for validation and is much simpler (we should avoid unnecessary checks).

```
1. `isset($a['key'])`:
- It's a language construct (not a function)
- Optimized at the language level
- Directly checks if the key exists and is not null
- Single operation

2. `$a !== []`:
- Requires comparison operation
- Needs to compare the entire array structure
- More complex operation as it checks if the array is not empty
```

---

@donhardman 

Looks like there's a flaw in this logic:

> You have declaration
> private array $customMapping;
> In case it returns false and we try to assign, we will fail already before this code, so I guess [] can be removed

Yes, it will fail if `false` is assigned to `$customMapping`, but `false` != `[]` and if `[]` is assigned to $customMapping, it won't fail, so:

```
			if ($this->customMapping !== []
				&& isset($this->customMapping[$fieldName])) {
```

seems to make sense as a protection from an **empty** array which is not the same as `false`.

---

The issue here is false never can be assigned to property with declared type array, this is fatal type error, we never will be able to see false in array

---

I agree `isset($this->customMapping[$fieldName])` is enough. `$this->customMapping !== []` is redundant.